plugins/action_update_review_status/src/updateReviewsStatuses.ts

```diff
@@ -18,38 +24,91 @@ const updateReviewsStatuses: ActionPluginType = async ({
+      : 'application Id: ' + applicationId
+  )
+
+  console.log('Current review level:', currentReviewLevel)
+  console.log('Changed responses:', changedResponses)
 
   const reviewsToUpdate = []
 
-  console.log('changedApplicationResponses', changedApplicationResponses)
+  const getReviewsByLevelAndStatus = async (
+    level: number,
+    statusToUpdate?: ReviewStatus[]
+  ): Promise<Review[]> =>
+    (await db.getAssociatedReviews(applicationId, stageId, level)).filter(
+      (review: Review) =>
+        review.reviewId !== reviewId &&
+        (!statusToUpdate || statusToUpdate.includes(review.reviewStatus))
+    )
+
+  console.log('Finding reviews to update status...')
 
   try {
-    // Get reviews/review assignments (with status) matching application_id & current stage
-    const reviews = (
-      await db.getAssociatedReviews(applicationId, stageId, level)
-    ).filter((review: Review) =>
-      [ReviewStatus.Submitted, ReviewStatus.Locked, ReviewStatus.Draft].includes(
-        review.reviewStatus
-      )
-    )
-    // Deduce which ones should be updated
-    for (const review of reviews) {
-      const { reviewAssignmentId, levelNumber, reviewStatus } = review
-      if (levelNumber > 1) reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (await haveAssignedResponsesChanged(reviewAssignmentId))
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (reviewStatus === ReviewStatus.Locked)
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
+    if (triggeredBy === 'REVIEW') {
+      // Review submitted from upper level to lower level review
+      if (decision === Decision.ChangesRequested) {
+        const previousLevelReview = currentReviewLevel - 1
+        // Update lower level reviews with assigned responses on the array changed responses ...
+        const reviewsSubmitted = await getReviewsByLevelAndStatus(previousLevelReview, [
+          ReviewStatus.Submitted,
+        ])
+        // Deduce which ones should be updated to Pending
+        for (const review of reviewsSubmitted) {
+          const { reviewAssignmentId } = review
+          if (await haveAssignedResponsesChanged(reviewAssignmentId))
+            reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.ChangesRequested })
+        }
+      } else {
+        // Review submitted from lower level to upper level
+
+        // Get all Locked reviews matching application_id, current stage, current level and in Locked status
+        const reviewsLocked = await getReviewsByLevelAndStatus(currentReviewLevel, [
+          ReviewStatus.Locked,
+        ])
+
+        // Now previous locked  to be allowed to continue
+        // Locked -> to avoid other reviews submitted while awaiting changes requests
+        reviewsLocked.forEach((review) =>
+          reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Draft })
+        )
```

Comment:
```suggestion

```

https://github.com/openmsupply/application-manager-server/pull/368#discussion_r636638083

Good to go after this one, thanks for changes

Reply:
Hi, sorry it wasn't clear on your previous suggestion. To me we would LOCK/UNLOCK reviews in the same level when another review gets CHANGE_REQUESTED by consolidator... I'm commenting the code now. And hopefully the confusion can be sorted on the next issue that I have in progress.